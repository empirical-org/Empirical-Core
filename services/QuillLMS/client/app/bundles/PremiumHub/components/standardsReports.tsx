import * as React from 'react';
import StandardsReportsTable from './standardsReportsTable';
import { ReportHeader } from '../../Shared';

export const StandardsReports = ({
  csvData,
  filteredStandardsReportsData,
  isFreemiumView
}) => {
  const freemiumClass = isFreemiumView ? 'freemium-view' : ''
  const header = isFreemiumView ? 'Premium Preview: Standards Report' : 'School Standards Reports'
  const tooltipText = isFreemiumView ? 'Subscribe to School or District Premium to unlock this report and more.' : "Each activity on Quill is aligned to a Common Core standard. This report shows the schoolâ€™s overall progress on each of the standards. You can print this report by downloading a PDF file or export this data by downloading a CSV file. The data you see below is capturing historical activity data for your school. <br/><br/> These reports are updated nightly."

  return(
    <div className={`admin-report-container standards-reports-by-classroom ${freemiumClass}`}>
      <ReportHeader
        className="admin-standards-report"
        csvData={csvData}
        headerText={header}
        isFreemiumView={isFreemiumView}
        tooltipText={tooltipText}
      />
      <StandardsReportsTable data={filteredStandardsReportsData} isFreemiumView={isFreemiumView} />
    </div>
  )
};

export default StandardsReports;
