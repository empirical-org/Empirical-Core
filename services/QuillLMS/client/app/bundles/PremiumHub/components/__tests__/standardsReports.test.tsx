import * as React from 'react';
import { shallow } from 'enzyme';

import { StandardsReports } from '../standardsReports';

describe('StandardsReports component', () => {
  const mockProps = {
    csvData: [],
    filteredStandardsReportsData: [],
  }
  const component = shallow(<StandardsReports {...mockProps} />);

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
