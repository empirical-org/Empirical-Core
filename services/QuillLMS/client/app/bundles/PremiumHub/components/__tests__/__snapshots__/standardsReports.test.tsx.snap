// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardsReports component should match snapshot 1`] = `
<div
  className="admin-report-container standards-reports-by-classroom "
>
  <ReportHeader
    className="admin-standards-report"
    csvData={Array []}
    headerText="Standards Report"
    tooltipText="Each activity on Quill is aligned to a Common Core standard. This report shows the schoolâ€™s overall progress on each of the standards. You can print this report by downloading a PDF file or export this data by downloading a CSV file. The data you see below is capturing historical activity data for your school. <br/><br/> These reports are updated nightly."
  />
  <StandardsReportsTable
    data={Array []}
  />
</div>
`;
